spec/unit/driver/auto_scaling_group_spec.rb

```diff
@@ -27,83 +27,152 @@
 
   describe '#warm_up' do
     it 'should warm up the group to the desired size' do
-      expect(group).to receive(:auto_scaling_instances){[instance1, instance2]}
       expect(group).to receive(:set_desired_capacity).with(2)
-      expect(group).to receive(:desired_capacity).and_return(2)
+      expect(@driver).to receive(:wait_for_desired_capacity)
       @driver.warm_up 2
     end
 
     it 'should wait for the warm up of the group even if desired is the same as the minimum' do
-      expect(group).to receive(:auto_scaling_instances){[instance2]}
       expect(group).to receive(:set_desired_capacity).with(1)
-      expect(group).to receive(:desired_capacity).and_return(1)
+      expect(@driver).to receive(:wait_for_desired_capacity)
       @driver.warm_up 1
     end
 
     it 'should ignore warming up if desired number is less than min size of the group' do
       expect(group).not_to receive(:set_desired_capacity)
-      expect(group).not_to receive(:desired_capacity)
+      expect(@driver).not_to receive(:wait_for_desired_capacity)
       @driver.warm_up 0
     end
 
     it 'should warm up to maximum if desired number is greater than maximum size of group' do
-      expect(group).to receive(:auto_scaling_instances){[instance1, instance2, instance3, instance4]}
       expect(group).to receive(:set_desired_capacity).with(4)
-      expect(group).to receive(:desired_capacity).and_return(4)
+      expect(@driver).to receive(:wait_for_desired_capacity)
       @driver.warm_up 5
     end
   end
 
-  describe '#healthy_instance_count' do
-    it 'should respond with the number of instances that are HEALTHY' do
-      instance5 =  double('instance1', :health_status => 'UNHEALTHY')
-      allow(group).to receive(:auto_scaling_instances){[instance1, instance2, instance3, instance4, instance5]}
-      expect(@driver.send(:healthy_instance_count)).to eql 4
```

Comment:
For reference, this is the existing test case that highlights the use case of an ASG with no associated ELB.